sway-lib-std/src/storage.sw

```diff
@@ -236,6 +236,31 @@ impl<V> StorageVec<V> {
         element_to_be_removed
     }
 
+    /// Sets/mutates the value at the given index
+    ///
+    /// # Arguments
+    /// 
+    /// * `index` - The index of the vec to set the value at
+    /// * `value` - The value to be set
+    ///
+    /// # Reverts
+    /// 
+    /// Reverts if index is larger than the length of the vec
+    #[storage(read, write)]
+    pub fn set(self, index: u64, value: V) {
+        let len = get::<u64>(__get_storage_key());  
+        // if the index is higher than len, cannot set an element.
+        assert(index <= len);
```

Comment:
Isn't it better to only allow setting indices that are within range? I don't expect this function to change the length of the vector  (unlike `push` and `insert`).

Also, soliciting feedback from @sezna @otrho @mitchmindtree regarding adding a function `set()` that I don't believe Rust has (but Rust has mutable references which we don't have yet). If we had `set()` here we also have to add `set()` in `Vec`.

Reply:
Well you could make the same argument for insert. If there nothing to insert between why do we allow it? But we still do. Itd be inconsistent to allow it for insert and not for set IMO

Reply:
Hmm yeah I see where you're coming from. My argument is that `insert()` always increases the length by 1 but `set()` does not. A function called `insert()` is always expected to grow the size because it's inserting something somewhere, and the end of a vector is a possible place for where the new element could go. A function called `set()` however is expected to set (aka change, update, etc.) an existing value. In the case of `Vec()` for example, I do not expect `set()` to ever incur allocation cost. Also, `set()` is meant to replace updating a mutable reference, and you can't really get a valid mutable reference to index `len()`.

Reply:
hmm yeah i see what you mean, ill change that